src/Generator.Bind/CSharpSpecWriter.cs

```diff
@@ -662,15 +662,13 @@ private string GetDeclarationString(Function f, Settings.Legacy settings)
             if (f.Parameters.HasGenericParameters)
             {
                 sb.Append("<");
-                foreach (Parameter p in f.Parameters)
```

Comment:
Worth replacing all this with `sb.Append(String.Join(", ", f.Parameters.Where(p => p.Generic))`?
Might need to add ToArray or change Bind to target .NET4.5 as well.